### PR TITLE
feat(Evaluator): support auto render as .mp4 videos, add examples and tests

### PR DESCRIPTION
**Pull request by muchvo**
## Description

1.Support directly render on server and save episodes as .mp4 videos.
2.Add new tests.
3.Add examples to show how to use.
4.Update some wrappers.
5.After carefully testing.
image

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format`. (**required**)
- [ ] I have checked the code using `make lint`. (**required**)
- [ ] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: envs/safety-gymnasium/setup.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env python3
```
Comment (zmsn-2077) [inline envs/safety-gymnasium/setup.py:1]: why do we need to add this line?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -45,6 +45,10 @@ jobs:
         run: |
           python -m pip install -vvv --editable '.[lint]'
 
+      - name: Install safety_gymnasium
```
Comment (zmsn-2077) [inline .github/workflows/ci.yml:48]: Why is this part modified? Does it need to be updated with the main branch?

**Review by zmsn-2077 — CHANGES_REQUESTED** [review]
.

**Review by zmsn-2077 — APPROVED** [review]
After careful review, I agree with these changes.

**Review by calico-1226 — APPROVED** [review]
I approve these changes.

**Pull request merged**